src/databases.py

```diff
@@ -11,16 +11,13 @@
 import yaml
 from charms.data_platform_libs.v0.data_interfaces import DatabaseRequires, DatabaseRequiresEvent
 
-from constants import (
-    FLASK_CONTAINER_NAME,
-    FLASK_DATABASE_NAME,
-    FLASK_SERVICE_NAME,
-    FLASK_SUPPORTED_DB_INTERFACES,
-)
-from exceptions import InvalidDatabaseRelationDataError, PebbleNotReadyError
+from constants import FLASK_DATABASE_NAME, FLASK_SUPPORTED_DB_INTERFACES
 
 logger = logging.getLogger(__name__)
 
+if typing.TYPE_CHECKING:
+    from charm import FlaskCharm
```

Comment:
Is the if statement needed?
It seems the import is only used for type checking, but in the team's code base we generally do not do this.

Reply:
This is necessary to avoid circular dependency.

Reply:
I think we should include a comment explaining this, as otherwise it might not be obvious to someone looking at it in the future.

Reply:
Yes, we will have a meeting discussing the circular dependency problem later this day, and hope we can remove them.

Reply:
This looks similar (exactly?) to what happened in Synapse where we changed for passing a Pebble instance. I see that you are doing this because the charm needs to call config_change ->  _restart_flask_application. Maybe it's necessary to add the same abstraction here.

Reply:
This has been removed in the latest commit, thanks!